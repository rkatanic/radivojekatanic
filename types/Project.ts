export interface ProjectType {
  slug: string;
  title: string;
  description: string;
  creationDate: string;
  content: string;
  updated: ProjectUpdated;
  links: ProjectLink[];
  icon?: string;
}

export interface ProjectUpdated {
  description: string;
  date: string;
}

export interface ProjectLink {
  title: string;
  href: string;
  type: ProjectLinkType;
}

export type ProjectLinkType = "application" | "image" | "readme";
