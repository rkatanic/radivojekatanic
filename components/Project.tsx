import { FiArrowRight } from "react-icons/fi";
import { DATE_FORMAT } from "utils/dateTimeUtils";
import { ProjectType } from "../types/Project";
import { TbSeeding } from "react-icons/tb";

interface Props {
  project: ProjectType;
}

const Project = ({
  project: { title, description, creationDate },
}: Props): JSX.Element => (
  <div className="flex gap-4 border-b border-slate-800 pb-8">
    <div className="flex h-14 w-14 items-center justify-center rounded-full border border-slate-800 bg-gradient-to-t from-slate-800/70 to-slate-800/10">
      <TbSeeding size="1.25rem" className="stroke-sky-500" />
    </div>
    <div className="flex-1">
      <p className="mb-1 text-sm text-slate-500">
        {DATE_FORMAT.format(new Date(creationDate))}
      </p>
      <h3 className="font-serif text-lg font-semibold tracking-wide text-slate-200">
        {title}
      </h3>
      <p className="mt-2">{description}</p>
      {/* <button className="border-tran mt-4 flex items-center gap-2 border-b border-transparent text-sm text-sky-400 hover:border-sky-500">
        View more <FiArrowRight />
      </button> */}
    </div>
  </div>
);

export default Project;
