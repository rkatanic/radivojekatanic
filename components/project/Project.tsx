import Link from "next/link";
import { FiChevronRight } from "react-icons/fi";
import { DATE_FORMAT } from "utils/dateTimeUtils";
import { ProjectType } from "../../types/Project";

interface Props {
  project: ProjectType;
  index: number;
}

const Project = ({
  project: { slug, title, description, creationDate },
  index,
}: Props): JSX.Element => (
  <Link className="flex w-full" href={`/projects/${slug}`}>
    <div className="group relative flex border border-gray-300 bg-slate-50 shadow-sm transition-[border] duration-200 hover:border-sky-400 dark:border-gray-700 dark:bg-gray-900 dark:shadow-md dark:hover:border-sky-400">
      <div className="flex-1 p-4">
        <div className="mb-4 text-xl font-medium tracking-wide dark:border-gray-700 dark:text-gray-200">
          {title}
        </div>
        <div className="absolute top-6 right-4 z-10 flex items-end gap-1 border border-gray-300 py-0.5 px-1 text-xs font-light tracking-wide text-gray-600 dark:border-gray-700 dark:text-gray-300">
          <span className="text-gray-400">#</span> 0{index + 1}
        </div>
        <div className="absolute bottom-6 right-4 z-10 flex items-end gap-1 text-sm font-light tracking-wide text-gray-400 group-hover:text-sky-500 dark:text-gray-500 dark:group-hover:text-sky-300">
          View <FiChevronRight className="mb-0.5" />
        </div>
        <p className="text-lg leading-relaxed tracking-wide text-gray-500 dark:text-gray-400">
          {description}
        </p>
        <hr className="my-4 dark:border-gray-800" />
        <div className="mb-2 flex items-center gap-2 text-sm text-gray-400 dark:text-gray-400">
          {DATE_FORMAT.format(new Date(creationDate))}
        </div>
      </div>
      <div className="xbg-gradient-to-r relative h-full w-44 overflow-hidden from-gray-900 to-gray-800/40">
        <div className="cube group-hover:animate-[rotate_30s_linear_infinite]">
          <div className="sphere-one inner"></div>
          <div className="sphere-two inner"></div>
          <div className="sphere-three inner"></div>
          <div className="sphere-four inner"></div>
          <div className="sphere-five inner"></div>
          <div className="sphere-six inner"></div>
        </div>
      </div>
    </div>
  </Link>
);

export default Project;
