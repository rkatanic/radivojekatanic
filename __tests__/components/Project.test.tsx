import Project from "@/components/Project";
import { render } from "@testing-library/react";

describe("Project", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(
      <Project
        project={{
          name: "Project name",
          description: "Project description.",
          creationDate: "2022-01-01",
          link: "project-url",
        }}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should render with badge", (): void => {
    const { baseElement } = render(
      <Project
        project={{
          name: "Project name",
          description: "Project description.",
          creationDate: "2022-01-01",
          link: "project-url",
          badgeLabel: "badge label",
        }}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
