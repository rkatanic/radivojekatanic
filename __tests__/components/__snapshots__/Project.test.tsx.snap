// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should render 1`] = `
<body>
  <div>
    <a
      class="border dark:border-neutral-800 w-full flex flex-col gap-4 p-6 rounded-xl hover:bg-neutral-50/50 dark:hover:bg-neutral-800/30 cursor-pointer group"
      href="project-url"
      target="blank"
    >
      <div
        class="flex items-baseline justify-between gap-4"
      >
        <h3
          class="flex flex-wrap gap-3 items-baseline font-medium dark:text-neutral-200 text-xl font-medium"
        >
          Project name
        </h3>
        <div
          class="whitespace-nowrap flex gap-3 items-center text-neutral-400 dark:text-neutral-600 text-sm"
        >
          Jan 01, 2022
        </div>
      </div>
      <p
        class="text-neutral-500 dark:text-neutral-400 mb-2"
      >
        Project description.
      </p>
      <hr
        class="dark:border-neutral-800"
      />
      <div
        class="gap-1 text-sm items-center text-neutral-400 dark:text-neutral-500 group-hover:text-teal-500 dark:group-hover:text-teal-300 flex"
      >
        Go to link
        <svg
          class="stroke-neutral-400 dark:stroke-neutral-500 group-hover:stroke-teal-400 dark:group-hover:stroke-teal-300"
          fill="none"
          height="1rem"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="7"
            x2="17"
            y1="17"
            y2="7"
          />
          <polyline
            points="7 7 17 7 17 17"
          />
        </svg>
      </div>
    </a>
  </div>
</body>
`;

exports[`Project should render with badge 1`] = `
<body>
  <div>
    <a
      class="border dark:border-neutral-800 w-full flex flex-col gap-4 p-6 rounded-xl hover:bg-neutral-50/50 dark:hover:bg-neutral-800/30 cursor-pointer group"
      href="project-url"
      target="blank"
    >
      <div
        class="flex items-baseline justify-between gap-4"
      >
        <h3
          class="flex flex-wrap gap-3 items-baseline font-medium dark:text-neutral-200 text-xl font-medium"
        >
          Project name
          <div
            class="whitespace-nowrap text-xs rounded-full bg-teal-50/50 dark:bg-teal-800/20 border border-teal-300 text-teal-600 dark:text-teal-300 max-w-min px-1.5"
          >
            badge label
          </div>
        </h3>
        <div
          class="whitespace-nowrap flex gap-3 items-center text-neutral-400 dark:text-neutral-600 text-sm"
        >
          Jan 01, 2022
        </div>
      </div>
      <p
        class="text-neutral-500 dark:text-neutral-400 mb-2"
      >
        Project description.
      </p>
      <hr
        class="dark:border-neutral-800"
      />
      <div
        class="gap-1 text-sm items-center text-neutral-400 dark:text-neutral-500 group-hover:text-teal-500 dark:group-hover:text-teal-300 flex"
      >
        Go to link
        <svg
          class="stroke-neutral-400 dark:stroke-neutral-500 group-hover:stroke-teal-400 dark:group-hover:stroke-teal-300"
          fill="none"
          height="1rem"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="7"
            x2="17"
            y1="17"
            y2="7"
          />
          <polyline
            points="7 7 17 7 17 17"
          />
        </svg>
      </div>
    </a>
  </div>
</body>
`;
